File: PessoaCRUD/java/models/Pessoa.java
package models;



import java.io.Serializable;
import java.text.ParseException;

import javax.persistence.Column;
import javax.persistence.Convert;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.swing.text.MaskFormatter;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

@Entity(name = "PESSOA")
public class Pessoa implements Serializable{
	@Id
	@GeneratedValue
	private int id;
	@Column(nullable=false, length=14)
	private String CPF;
	@Column(nullable=false, length=45)
	private String nome;
	@Column(nullable=false, length=14)
	private String telefone;
	@Column(nullable=false, length=45)
	@Convert(converter = EscolaridadeConverter.class)
	private Escolaridade escolaridade;
	
	public Pessoa() {
		
	}
	
	@JsonCreator
	public Pessoa(
			@JsonProperty("CPF") String CPF, 
			@JsonProperty("nome") String nome, 
			@JsonProperty("telefone") String telefone,
			@JsonProperty("escolaridade") Escolaridade escolaridade
			) {
		this.setCPF(CPF);
		this.setNome(nome);
		this.setTelefone(telefone);
		this.setEscolaridade(escolaridade);
	}
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id=id;
	}
	@JsonProperty("CPF")
	public String getCPF() {
		return CPF;
	}
	public void setCPF(String CPF) {
		if(CPF.matches("[0-9]{3}.[0-9]{3}.[0-9]{3}-[0-9]{2}")) {
			this.CPF = CPF;
		}else {
			throw new IllegalArgumentException("CPF Inválido");
		}
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		if(!nome.isEmpty()) {
			this.nome = nome;
		}else {
			throw new IllegalArgumentException("Informe o nome");
		}
	}

	public String getTelefone() {
		return telefone;
	}

	public void setTelefone(String telefone) {
		if(telefone.matches("\\([0-9]{2}\\)([0-9]{5}|[0-9]{4})-[0-9]{4}")) {
			this.telefone=telefone;
		}else {
			throw new IllegalArgumentException("Telefone Inválido");
		}	
	}

	public Escolaridade getEscolaridade() {
		return escolaridade;
	}

	public void setEscolaridade(Escolaridade escolaridade) {
		this.escolaridade = escolaridade;
	}
	@Override
	public int hashCode() {
		return id/7;
	}
	public boolean equals(Object o) {
		Pessoa pessoa = (Pessoa)o;
		return this.id==pessoa.id;
	}
}

File: PessoaCRUD/src/main/webapp/JS/Pessoa.js
var Escolaridade;
(function (Escolaridade) {
    Escolaridade["FUNDAMENTAL"] = "Fundamental";
    Escolaridade["MEDIO"] = "M\u00E9dio";
    Escolaridade["SUPERIOR_INCOMPLETO"] = "Superior Incompleto";
    Escolaridade["SUPERIOR_COMPLETO"] = "Superior Completo";
    Escolaridade["TECNICO_INCOMPLETO"] = "T\u00E9cnico Incompleto";
    Escolaridade["TECNICO_COMPLETO"] = "T\u00E9cnico Completo";
})(Escolaridade || (Escolaridade = {}));
var Pessoa = /** @class */ (function () {
    function Pessoa(id, cpf, nome, telefone, escolaridade) {
        this.setId(id);
        this.setCpf(cpf);
        this.setNome(nome);
        this.setTelefone(telefone);
        this.setEscolaridade(escolaridade);
    }
    Pessoa.createFromAnyObject = function (obj) {
        return new Pessoa(obj["id"], obj["CPF"], obj["nome"], obj["telefone"], obj["escolaridade"]);
    };
    Pessoa.prototype.getId = function () {
        return this.id;
    };
    Pessoa.prototype.setId = function (id) {
        this.id = id;
    };
    Pessoa.prototype.getCpf = function () {
        return this.cpf;
    };
    Pessoa.prototype.getNome = function () {
        return this.nome;
    };
    Pessoa.prototype.getTelefone = function () {
        return this.telefone;
    };
    Pessoa.prototype.getEscolaridade = function () {
        return this.escolaridade;
    };
    Pessoa.prototype.setCpf = function (cpf) {
        if (cpf.length == 0) {
            throw new Error("Informe o CPF");
        }
        else {
            if (cpf.match("[0-9]{3}.[0-9]{3}.[0-9]{3}-[0-9]{2}")) {
                this.cpf = cpf;
            }
            else {
                throw new Error("CPF Inválido");
            }
        }
    };
    Pessoa.prototype.setNome = function (nome) {
        console.log(nome.length);
        if (nome.length == 0) {
            throw new Error("Informe o nome");
        }
        else {
            this.nome = nome;
        }
    };
    Pessoa.prototype.setTelefone = function (telefone) {
        if (telefone.length == 0) {
            throw new Error("Informe o Telefone");
        }
        else {
            if (telefone.match("\\([0-9]{2}\\)([0-9]{5}|[0-9]{4})-[0-9]{4}")) {
                this.telefone = telefone;
            }
            else {
                throw new Error("Telefone Inválido");
            }
        }
    };
    Pessoa.prototype.setEscolaridade = function (escolaridade) {
        this.escolaridade = escolaridade;
    };
    Pessoa.prototype.toRaw = function () {
        return "<tr><td>" + this.cpf + "</td><td>" + this.nome + "</td><td class=\"telcolumn\">" + this.telefone + "</td><td class=\"escolcolumn\">" + Escolaridade[this.escolaridade] + "</td><td><input type=\"radio\" name=\"selecao\" id=\"selecao\" class=\"selecao\" value=\"" + this.id + "\"/></td></tr>";
    };
    Pessoa.prototype.toJson = function () {
        return "{ \"CPF\" : \"" + this.cpf + "\", \"nome\" : \"" + this.nome + "\", \"telefone\" : \"" + this.telefone + "\", \"escolaridade\" : \"" + this.escolaridade + "\" }";
    };
    Pessoa.prototype.toJsonWithId = function () {
        return "{ \"id\" : \"" + this.id + "\", \"CPF\" : \"" + this.cpf + "\", \"nome\" : \"" + this.nome + "\", \"telefone\" : \"" + this.telefone + "\", \"escolaridade\" : \"" + this.escolaridade + "\" }";
    };
    return Pessoa;
}());
var PessoaJSONSerializable = /** @class */ (function () {
    function PessoaJSONSerializable() {
    }
    PessoaJSONSerializable.prototype.CreateObjectFromJSON = function (data) {
        return new Pessoa(data["id"], data["CPF"], data["nome"], data["telefone"], data["escolaridade"]);
    };
    PessoaJSONSerializable.prototype.CreateArrayFromJSON = function (data) {
        var i;
        var pessoas = new Array();
        for (i = 0; i < data.length; i++) {
            pessoas[i] = this.CreateObjectFromJSON(data[i]);
        }
        return pessoas;
    };
    return PessoaJSONSerializable;
}());
